appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb

```diff
@@ -238,7 +272,46 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "expr_df.head()"
+    "# check if data have been loaded correctly\n",
+    "try:\n",
+    "    pca = PCA(n_components=clustering_topk)\n",
+    "    pca.fit_transform(expr_df)\n",
+    "except:\n",
+    "    raise Exception(\"Error! Incorrect format. Please check if the data have been read/seperated correctly.\")"
```

Comment:
Any reason you do this twice if normalizing data?

I think you can just take the PCA out of the if needs_normalize below

Reply:
Sometimes when input data has been normalized, we get many NA values after normalization, so I was checking if the input data has already been normalized.